week3/src/components/CardContainer.jsx

```diff
@@ -0,0 +1,145 @@
+            e.status = !e.status
+            return true
+        }
+          return false
+        })
+    
+        setClicked([...clicked, idx])
+      }
+    
+      useEffect(() => {
+        setTimeout(() => {
+          if (clicked.length === 2) {
+            let a = MixedCardList.find((e) => e.idx === clicked[0]).id
+            let b = MixedCardList.find((e) => e.idx === clicked[1]).id
+            
+            // 다른 경우 status를 false로 유지
+            if (a !== b) {
+              MixedCardList.forEach((e) => {
+                if (e.idx === clicked[0] || e.idx === clicked[1]) {
+                  e.status = false
+                }
+              })
+            } else{
+              // 같은 경우 점수 올리기
+              updateScore((currentScore) => currentScore + 1);
+            }
+            // 선택 끝나면 배열 비우기
+            setClicked(Array.from([]))
+          }
+        }, 700)
+      }, [clicked])
+    
+      useEffect(() => {
+        if (MixedCardList.every((e) => e.status === true)) {
+          setFinished(true)
+        }
+      }, [clicked])
+    
+      const handleFinished = () => {
+        setFinished(false)
+        updateScore(0);
+      }
+
+    return(
+        <GameLayout>
+            {finished && (
+            <ClearModal handleFinished={handleFinished} >
+            </ClearModal>
+      )}
+            <CardsLayout>
+                {MixedCardList.map((e) => (
+                <Card
+                    data={e}
+                    key={e.idx}
+                    handleClick={clicked.length < 2 ? handleClick : null}
+                />
+                ))}
+            </CardsLayout>
+        </GameLayout>
+        
+    )
+}
+
+const GameLayout = styled.div`
+    display: flex;
+    width: 100%;
+    height: 37rem;
+    justify-content: center;
+    background-color: ${({ theme }) => theme.colors.KBOLightBlue};
+`
+
+const CardsLayout = styled.section`
+    display: grid;
+    grid-template-columns: repeat(8, 1fr);
+`
+
+
+function MixedCardDeck(){
+    let cardDeck = [];
+    let randomNumberArr = [];
+
+    // 여기는 난이도 개수만큼
+    for (let i = 0; i < LEVEL.EASY; i++) {
+        // 랜덤 값 뽑기
+        let randomNumber = getRandom(10, 0);    // 이미지 개수만큼
+    
+        // 중복 검사
+        if (randomNumberArr.indexOf(randomNumber) === -1) {
+            randomNumberArr.push(randomNumber);
+        } else {
+            i--;
+        }
+    }
+
+    // 카드는 두 장씩 필요 >> 한 번 더 추가
+    randomNumberArr.push(...randomNumberArr);
+
+    // 카드 섞기
+    shuffle(randomNumberArr);
+
+    // 섞은 값으로 카드 세팅
+    for (let i = 0; i < LEVEL.EASY * 2; i++) {
+        cardDeck.push({
+            id: CARDS[randomNumberArr[i]].id,
+            name: CARDS[randomNumberArr[i]].name, 
+            img: CARDS[randomNumberArr[i]].img, 
+            status: false,
+            idx: i
+        });
+    }
+
+    return cardDeck;
+}
+
+// 난수 생성 함수
+function getRandom(max, min) {
+    return parseInt(Math.random() * (max - min)) + min;
+}
```

Comment:
지금 이 로직을 보았을 때, min~max -1 범위에서 난수를 생성해서 리턴하는 함수 같아요!!

그런데 이 함수를 사용하는 곳을 보니까 0부터 9까지 난수를 뽑는 용도로 사용되더라구요!! 위에 사용하는 로직을 보니 0부터 이미지 개수 - 1 까지 난수를 발생시켜 이미지의 위치를 잡는 용도로 이해했어요!!

그렇다면, 난이도가 늘어나더라도 0부터 이미지 개수 - 1까지인건 동일할 것 같은데, 이렇다면 min이 0으로 고정이니 매개변수 1개로 더 간단하게 구현이 가능할 것 같다는 생각이 들었어요!!

```suggestion
function getRandom(imgCount) {
    return Math.floor(Math.random() * imgCount);
}
```

Reply:
오 확실히 깔끔해졌네요!